unittests/compiled_tests.cpp

```diff
@@ -352,6 +352,16 @@ TEST_CASE("Functor cast")
   CHECK(d == 3 * 6);
 }
 
+TEST_CASE("BOM at beginning of string")
+{
+  chaiscript::ChaiScript_Basic chai(create_chaiscript_stdlib(),create_chaiscript_parser());
+
+  chai.add(chaiscript::fun(&functor_cast_test_call), "test_call");
+
+  chai.eval("def func() { return \"Hello World\"; };");
+
+  CHECK(chai.eval<std::string>("\xef\xbb\xbf(func())") == "Hello World");
```

Comment:
IMO this whole script should be inline (i.e. don't use `func()`), since you want to test the script running, not the function call working. Also you should probably remove line 359 as well.